app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -206,6 +207,23 @@ public StudyParticipant getParticipant(Study study, Account account, boolean inc
                 consentHistories.put(subpop.getGuidString(), history);
             }
             builder.withConsentHistories(consentHistories);
+
+            // To calculate consent status, we need construct a CriteriaContext from RequestInfo.
+            RequestInfo requestInfo = cacheProvider.getRequestInfo(account.getId());
+            if (requestInfo != null) {
+                CriteriaContext criteriaContext = new CriteriaContext.Builder()
+                        .withStudyIdentifier(study.getStudyIdentifier())
+                        .withUserId(account.getId())
+                        .withHealthCode(account.getHealthCode())
+                        .withClientInfo(requestInfo.getClientInfo())
+                        .withLanguages(requestInfo.getLanguages())
+                        .withUserDataGroups(requestInfo.getUserDataGroups())
+                        .build();
+                Map<SubpopulationGuid, ConsentStatus> consentStatusMap = consentService.getConsentStatuses(
+                        criteriaContext, account);
+                boolean isConsented = ConsentStatus.isUserConsented(consentStatusMap);
+                builder.withConsented(isConsented);
+            }
```

Comment:
Created a JIRA to persist RequestInfo outside of Redis, and ensure it's always 100% a reflection of what we're using to calculate content returned to the user: https://sagebionetworks.jira.com/browse/BRIDGE-2284